File: Python/03-cuarto/02-script.py
#!/usr/bin/env python3

# Programa una funcion que determine si un numero es par o impar. miFuncion(29)
# devolvera Impar

def es_par(numero):
    if numero % 2 == 0:
        return True
    return False

def main():
    numero = input("Ingresa un numero => ")
    assert numero.isnumeric(), "Debes ingresar un numero"
    numero = int(numero)  # Convertir la entrada a un entero
    if es_par(numero):
        print("Es par")
    else:
        print("Es impar")

if __name__ == '__main__':
    main()

File: JavaScript/03-cuarto/02-script.js
/*
Programa una funcion que determine si un numero es par o impar. miFuncion(29)
devolvera Impar */

const es_par = (numero) => {
    if (!numero) return console.warn("Debes ingresar un numero");

    return (numero % 2)
        ? console.log(`El número ${numero}, es impar`)
        : console.log(`El número ${numero}, Sí es par`);
}

// es_par();
// es_par(29);
// es_par(4);
// es_par(7);


//////////////////////////////////////////////////////////////////

// solucion jhon mircha

const numeroParImpar = (numero = undefined) => {
    if (numero === undefined) return console.warn("No ingresaste un numero");

    if (typeof numero != "number") return console.error(`El valor ${numero} ingresado, No es un numero`);

    return ((numero % 2) === 0)
        ? console.info(`El numero ${numero} es Par`)
        : console.info(`El numero ${numero} es Impar`);
}

numeroParImpar();
numeroParImpar("54");
numeroParImpar(-398);
numeroParImpar(19);